### ISSUE TITLE
Process leak in webmachine access logging

### ISSUE THREAD

**Issue by jcapricebasho** [webmachine/webmachine#183]
Discovered in Riak 1.4.6 with the following app.config line used to enable webmachine access logging:

```
{webmachine, [{log_handlers, [{webmachine_access_log_handler, ["/var/log/riak/"]}]}]},
```

process_info of an offending process:

```
{<0.28899.3>,
              {<0.28899.3>,#Ref<0.0.2.12594>},
              {sync_notify,{log_access,{wm_log_data,riak_kv_wm_object,
                                                    {1389,899062,42961},
                                                    'POST',
                                                    {5,
                                                     {"content-length",
                                                      {'Content-Length',"24"},
                                                      {"accept",{'Accept',"*/*"},nil,nil},
                                                      {"content-type",
                                                       {'Content-Type',"text/plain"},
                                                       nil,
                                                       {"host",
                                                        {'Host',"127.0.0.1:10018"},
                                                        nil,
                                                        {"user-agent",
                                                         {'User-Agent',"curl/7.24.0 (x86_64-apple-darwin12.0) libcurl/7.24.0 OpenSSL/0.9.8y zlib/1.2.5"},
                                                         nil,nil}}}}},
                                                    "127.0.0.1","127.0.0.1",
                                                    "/buckets/mytest/keys/key9057?returnbody=true",
                                                    {1,1},
                                                    {200,undefined},
                                                    24,
                                                    {1389,899062,46194},
                                                    {1389,899062,46304},
                                                    []}}}},
```

backtrace of an offending process:

```
(dev1@127.0.0.1)10> io:format("~s\n", [element(2, process_info(c:pid(0,28899,3), backtrace))]).
Program counter: 0x000000001bdab350 (gen:do_call/4 + 576)
CP: 0x0000000000000000 (invalid)
arity = 0

0x000000002cf8b460 Return addr 0x000000001c707a70 (gen_event:rpc/2 + 96)
y(0)     #Ref<0.0.2.12594>
y(1)     'dev1@127.0.0.1'
y(2)     []
y(3)     infinity
y(4)     {sync_notify,{log_access,{wm_log_data,riak_kv_wm_object,{1389,899062,42961},'POST',{5,{"content-length",{'Content-Length',"24"},{"accept",{'Accept',"*/*"},nil,nil},{"content-type",{'Content-Type',"text/plain"},nil,{"host",{'Host',"127.0.0.1:10018"},nil,{"user-agent",{'User-Agent',"curl/7.24.0 (x86_64-apple-darwin12.0) libcurl/7.24.0 OpenSSL/0.9.8y zlib/1.2.5"},nil,nil}}}}},"127.0.0.1","127.0.0.1","/buckets/mytest/keys/key9057?returnbody=true",{1,1},{200,undefined},24,{1389,899062,46194},{1389,899062,46304},[]}}}
y(5)     <0.28899.3>
y(6)     <0.193.0>

0x000000002cf8b4a0 Return addr 0x000000001baa2048 (<terminate process normally>)

ok
```

With the access log handler added, nodes will eventually crash hitting the process system_limit.

**Comment by jcapricebasho** [comment]
Message queue of webmachine_log_event appears to be blocked:

4442 =proc:<0.188.0>
4443 State: Garbing
4444 Name: webmachine_log_event
4445 Spawned as: proc_lib:init_p/5
4446 Spawned by: <0.187.0>
4447 Started: Wed Feb 12 15:28:19 2014
4448 Message queue length: 181380
4449 Number of heap fragments: 0
4450 Heap fragment data: 0
4451 Link list: [#Port<0.3349>, #Port<0.3325>, <0.192.0>, <0.191.0>, <0.187.0>, {from,<0.13031.12>,#Ref<0.0.18.196608>} ... ]
4452 Reductions: 48938635
4453 Stack+heap: 116769640
4454 OldHeap: 0
4455 Heap unused: 553
4456 OldHeap unused: 0
4457 Program counter: 0x00007f7caaa5cc28 (gen:do_call/4 + 576)
4458 CP: 0x0000000000000000 (invalid)

**Comment by nickelization** [comment]
This is now fixed, thanks to #197. I opened a remote shell and manually rotated the logs. Without the patch it deadlocked, and with the patch it succeeded.
